src/main/java/com/imsweb/mph/mpgroups/Mp2018HeadAndNeckGroup.java

```diff
@@ -40,45 +41,63 @@ public Mp2018HeadAndNeckGroup() {
-                else if ((GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.MAXILLARY_SINUS, MphConstants.ETHMOID_SINUS)) ||
-                        (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.MAXILLARY_SINUS, MphConstants.FRONTAL_SINUS)) ||
-                        (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.MAXILLARY_SINUS, MphConstants.SPHENOID_SINUS)) ||
-                        (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.ETHMOID_SINUS, MphConstants.FRONTAL_SINUS)) ||
-                        (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.ETHMOID_SINUS, MphConstants.SPHENOID_SINUS)) ||
-                        (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.FRONTAL_SINUS, MphConstants.SPHENOID_SINUS)))
+                else if ((GroupUtility.differentCategory(s1, s2, MphConstants.MAXILLARY_SINUS, MphConstants.ETHMOID_SINUS)) ||
+                        (GroupUtility.differentCategory(s1, s2, MphConstants.MAXILLARY_SINUS, MphConstants.FRONTAL_SINUS)) ||
+                        (GroupUtility.differentCategory(s1, s2, MphConstants.MAXILLARY_SINUS, MphConstants.SPHENOID_SINUS)) ||
+                        (GroupUtility.differentCategory(s1, s2, MphConstants.ETHMOID_SINUS, MphConstants.FRONTAL_SINUS)) ||
+                        (GroupUtility.differentCategory(s1, s2, MphConstants.ETHMOID_SINUS, MphConstants.SPHENOID_SINUS)) ||
+                        (GroupUtility.differentCategory(s1, s2, MphConstants.FRONTAL_SINUS, MphConstants.SPHENOID_SINUS)))
                     result.setFinalResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
                     //   • Submandibular gland C080 AND sublingual gland C081
-                else if (GroupUtility.differentCategory(i1.getPrimarySite(), i2.getPrimarySite(), MphConstants.SUBMANDIBULAR_GLAND, MphConstants.SUBLINGUAL_GLAND))
+                else if (GroupUtility.differentCategory(s1, s2, MphConstants.SUBMANDIBULAR_GLAND, MphConstants.SUBLINGUAL_GLAND))
+                    result.setFinalResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
+                    //Glottis C320 AND/OR supraglottis C321 AND/OR subglottis C322 AND/OR laryngeal cartilage C323
+                else if (MphConstants.GLOTTIS_AND_LARYNGEAL_SITES.containsAll(Arrays.asList(s1, s2)) && !s1.equals(s2))
+                    result.setFinalResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
+                    //Maxilla C410 AND Mandible C411
+                else if (GroupUtility.differentCategory(s1, s2, MphConstants.MAXILLA, MphConstants.MANDIBLE))
+                    result.setFinalResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
+                    // Postcricoid C130 AND/OR hypopharyngeal aspect of aryepiglottic fold C131 AND/OR posterior wall of hypopharynx C132
```

Comment:
You do not need to change anything in here, I just wanted to comment on it:
- having comments between the if statements messes makes it difficult to know if they go with the prior statement, or the next if. It's not a big deal, it's just not very readable, and there is probably other way to organize the code so that is more readable.
- I see here some comments with a "dot" in front of them; if you copied this form a Word document, please make sure you don't copy any Windows characters in the source code.
- this class (and probably many others) use MphConstants everywhere. You should consider using static imports for those constants, it would make the code more readable (again, no changes needed now, only suggestions for the future).

Reply:
Same here, I will probably do some clean up for the whole library some day.

Reply:
I removed the comments here,